### PR TITLE
SkipChain Manager 2/2

### LINKED ISSUE #859: add scmgr

### PR DESCRIPTION
**Pull request by ineiti**
Closes #859

### CODE REVIEW

File: scmgr/app.go [hunk 1]
```diff
@@ -0,0 +1,562 @@
...
+
+// Returns the number of calls.
+func add(c *cli.Context) error {
+	log.Info("Adding a block with a new group")
+	if c.NArg() < 2 {
+		return errors.New("Please give group-file and id to add")
+	}
+	group := readGroup(c, 1)
+	cfg := getConfigOrFail(c)
+	sb := cfg.Sbm.GetFuzzy(c.Args().First())
```
Comment (nikkolasg) [inline scmgr/app.go]: the joy of naming ;)

File: scmgr/app.go [hunk 2]
```diff
@@ -0,0 +1,562 @@
...
+	if sb == nil {
+		return errors.New("didn't find latest block in local store")
+	}
+	client := skipchain.NewClient()
+	guc, cerr := client.GetUpdateChain(sb.Roster, sb.Hash)
+	if cerr != nil {
+		return errors.New("while updating chain: " + cerr.Error())
+	}
+	if len(guc.Update) == 1 {
+		log.Info("No new block available")
```
Comment (nikkolasg) [inline scmgr/app.go]: cant you return here since there's nothing new ?

File: scmgr/app.go [hunk 3]
```diff
@@ -0,0 +1,562 @@
...
+		return errors.New("couldn't read config: " + err.Error())
+	}
+
+	// Get the list of genesis block
+	genesis := sbl{}
+	for _, sb := range cfg.Sbm.SkipBlocks {
+		if sb.Index == 0 {
+			genesis = append(genesis, sb)
+		}
+	}
```
Comment (nikkolasg) [inline scmgr/app.go]: Seems like this block of code is present multiple times. Might be good to factor it in its own utility method maybe ? As you deem fit.

File: scmgr/app.go [hunk 4]
```diff
@@ -0,0 +1,562 @@
...
+						if _, exists := sisAll[si.ID]; !exists {
+							log.Print("Adding", si)
+							sisNew[si.ID] = si
+							sisAll[si.ID] = si
+						}
+					}
+				}
+			}
+		}
+	}
```
Comment (nikkolasg) [inline scmgr/app.go]: All of the above brings quite some conditional complexity -> 6 inner conditions. It'd be best if it were reduced to 2 or 3, but if it does not make sense (i.e. it only adds new functions only being used here), then it's fine.

File: scmgr/app.go (hunk 2 above)
Comment (ineiti) [inline scmgr/app.go]: yes, but it might be useful to have the genesis-id, and hash of the last block anyway.

**Pull request merged**